src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String EXCEPTION_TEXT = "Index out of bound exception.";
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            Node<T> first = new Node<>(null, value, null);
+            head = first;
+            tail = first;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
```

Comment:
```suggestion
        Node<T> newNode = new Node<>(tail, value, null);
        if (size == 0) {
            head = first;
        } else {
            tail.next = newNode;
        }
        tail = newNode;
```
let's always try to move out the lines that repeat in both `if` and `else`